### PR TITLE
Scrutinizer support

### PR DESCRIPTION
**Pull request by nokitakaze**
As I said before in #442 we need Scrutinizer to see our **code coverage** and code quality.

After that PR the maintainer have to add branch 2.0 to "Tracked Branches"

### CODE REVIEW

File: .scrutinizer.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    php_analyzer: true
+
+build:
+    environment:
+        php:
+            version: 5.6.0
+    tests:
+        override:
+            -
+                command: 'vendor/bin/phpunit --verbose --configuration phpunit.xml --coverage-clover test/clover.xml'
```
Comment (ste93cry) [inline .scrutinizer.yml]: I never used Scrutinizer before, but I think this is going to execute again the unit tests after Travis. Isn't there a way to upload the `clover.xml` file from Travis to Scrutinizer? This way we would run the test suite only once

File: .scrutinizer.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                coverage:
+                    file: 'test/clover.xml'
+                    format: 'clover'
+    environment:
+        redis: false
+        postgresql: false
+        mongodb: false
+
+filter:
+    excluded_paths: [vendor/*, test/*, bin/*, docs/*, examples/*, test/*]
```
Comment (ste93cry) [inline .scrutinizer.yml]: `test/*` directory is specified twice

Comment (nokitakaze) [inline .scrutinizer.yml]: ooops

File: README.md [hunk 3]
```diff
@@ -11,6 +11,8 @@
 [![Downloads per month](https://img.shields.io/packagist/dm/sentry/sentry.svg?style=flat-square)](https://packagist.org/packages/sentry/sentry)
 [![Latest stable version](https://img.shields.io/packagist/v/sentry/sentry.svg?style=flat-square)](https://packagist.org/packages/sentry/sentry)
 [![License](http://img.shields.io/packagist/l/sentry/sentry.svg?style=flat-square)](https://packagist.org/packages/sentry/sentry)
+[![Scrutinizer Code Quality](https://scrutinizer-ci.com/g/getsentry/sentry-php/badges/quality-score.png?b=master)](https://scrutinizer-ci.com/g/getsentry/sentry-php/)
```
Comment (ste93cry) [inline README.md]: You should use [shields.io](http://shields.io/) like all other badges, and the same for the line below

File: .scrutinizer.yml (hunk 1 above)
Comment (nokitakaze) [inline .scrutinizer.yml]: > Isn't there a way to upload

No. It\'s not possible

Comment (ste93cry) [inline .scrutinizer.yml]: It is: look [here](https://adayinthelifeof.nl/2013/11/20/external-code-coverage-with-travis-scrutinizer/)

Comment (nokitakaze) [inline .scrutinizer.yml]: Ho Ho Ho! Nice. I'll change it in an hour.

Comment (nokitakaze) [inline .scrutinizer.yml]: It\'s funny. I don\'t know how code coverage works for version_compare PHP_VERSION. It\'s clear Clover can not see unreachable lines so he will think these lines are not coveraged

**Comment by nokitakaze** [comment]
Hm, I don\'t know why but Scrutinizer does not see uploaded files https://travis-ci.org/getsentry/sentry-php/builds/226327318

Inspection on my fork ended successfully

**Comment by nokitakaze** [comment]
I think thats because commits does not equals. Travis send coverage with commit 8ddde4d, but merging commit is c32f0c5bc

File: .travis.yml [hunk 4]
```diff
@@ -59,4 +59,12 @@ install: travis_retry composer install --no-interaction --prefer-source
 
 script:
   - composer phpcs
-  - composer tests
+  - composer tests-travis
+
+after_script:
+  - wget https://scrutinizer-ci.com/ocular.phar
+  - if [ $(phpenv version-name) = "5.3" ] && [ "$REMOVE_XDEBUG" = "0" ]; then php ocular.phar code-coverage:upload --format=php-clover test/clover.xml --revision=$TRAVIS_COMMIT; fi
```
Comment (ste93cry) [inline .travis.yml]: Why you duplicated the same command for each PHP version? If it's the same then there should be no need to do this

Comment (nokitakaze) [inline .travis.yml]: 7.1, HHVM

Comment (ste93cry) [inline .travis.yml]: 7.1 should be tested too because it's a stable PHP version

Comment (nokitakaze) [inline .travis.yml]: It **must** be tested on branch 2.0. But **must not** be testes on the branch where is failure available.

**Comment by nokitakaze** [comment]
After this pr will be merged we should make pull request this all renews from master branch to 2.0.

@dcramer, we wait you.

**Comment by nokitakaze** [comment]
Hooray. I don\'t know who is the founder of https://scrutinizer-ci.com/g/getsentry/sentry-php/ , but we need to add branch "2.0" to tracked branches https://scrutinizer-ci.com/g/getsentry/sentry-php/settings

We need to see how our code coverage and code quality changing in new version

**Comment by dcramer** [comment]
Added the branch

**Comment by nokitakaze** [comment]
> we should make pull request this all renews

Oookay. @ste93cry it will be me or you?

**Comment by ste93cry** [comment]
It's fine if you could do it, thank you

**Comment by nokitakaze** [comment]
> if you could do it

PR #466

**Pull request merged**